visualization/dashboard/src/MLIDashboard/Controls/ModelPerformanceTab/ModelPerformanceTab.tsx

```diff
@@ -0,0 +1,398 @@
+
+        this.props.onChange(newProps);
+        this.setState({xDialogOpen: false})
+    }
+
+    private onYSet(value: ISelectorConfig): void {
+        const newProps = _.cloneDeep(this.props.chartProps);
+        newProps.yAxis = value;
+
+        this.props.onChange(newProps);
+        this.setState({yDialogOpen: false})
+    }
+
+    private generateDefaultChartAxes(): void {
+        let bestModelMetricKey: string;
+        if (this.props.metadata.modelType === ModelTypes.binary && this.props.jointDataset.hasPredictedProbabilities) {
+            bestModelMetricKey = JointDataset.ProbabilityYRoot + "0";
+        } else if (this.props.metadata.modelType === ModelTypes.regression) {
+            if (this.props.jointDataset.hasPredictedY && this.props.jointDataset.hasTrueY) {
+                bestModelMetricKey = JointDataset.RegressionError;
+            } else {
+                bestModelMetricKey = JointDataset.PredictedYLabel;
+            }
+        } else {
+            bestModelMetricKey = JointDataset.PredictedYLabel;
+        } // not handling multiclass at this time
+
+        const chartProps: IGenericChartProps = {
+            chartType: this.props.jointDataset.metaDict[bestModelMetricKey].isCategorical ? ChartTypes.Bar : ChartTypes.Box,
+            yAxis: {
+                property: Cohort.CohortKey,
+                options: {}
+            },
+            xAxis: {
+                property: bestModelMetricKey,
+                options: {
+                    bin: false
+                }
+            }
+        }
+        this.props.onChange(chartProps);
+    }
+
+    private static generatePlotlyProps(jointData: JointDataset, chartProps: IGenericChartProps, cohorts: Cohort[], selectedCohortIndex: number): IPlotlyProperty {
+        // In this view, y will always be categorical (including a binned numberic variable), and could be
+        // iterations over the cohorts. We can set y and the y labels before the rest of the char properties.
+        const plotlyProps: IPlotlyProperty = {
+            config: { displaylogo: false, responsive: true, displayModeBar: false},
+            data: [{}],
+            layout: {
+                dragmode: false,
+                autosize: true,
+                margin: {
+                    l: 10,
+                    t: 25,
+                    b: 20,
+                },
+                hovermode: "closest",
+                showlegend: false,
+                yaxis: {
+                    automargin: true,
+                    color: FabricStyles.chartAxisColor,
+                    tickfont: {
+                        family: FabricStyles.fontFamilies,
+                        size: 11
+                    },
+                    showline: true
+                },
+                xaxis: {
+                    side: "bottom",
+                    mirror: true,
+                    color: FabricStyles.chartAxisColor,
+                    tickfont: {
```

Comment:
this kind of style stuff should live in styles?

Reply:
as above, this is a config, not css, but I understand the confusion. That said, it can be moved to a shared location if it is shared.